src/components/header/header.style.js

```diff
@@ -0,0 +1,27 @@
+import styled from 'styled-components/macro';
+import { Link } from 'react-router-dom';
+
+export const Container = styled.header`
+  height: 100px;
+  width: 100vw;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`;
+
+export const Logo = styled.img`
+  margin-left: 80px;
+  width: 150px;
+  height: 36px;
+`;
+
+export const NavLink = styled(Link)`
+  font-size: ${(props) => props?.theme?.font?.size?.default};
```

Comment:
It should be save to access the theme variables without `?`. It's defined by you in the end. So if something is missing it might be good to see an error immediately 😉

Reply:
I find this a little bit tricky. The theme actually works pretty well. I printed it out to the console and the theme object indeed showed up. But when it comes to testing, I got undefined. After doing some research, I move the `<ThemeProvider>` to the App.js and readjust the hierarchy, everything looks fine.

Reply:
Yeah exactly. That's the way to go. You always either need to include the context providers (e.g. ThemeProvider) into the component you render during the test as you did here. Or you wrap the component in the required providers inside the test. This is often done when you don't render the whole app component but only e.g. the home page or any other component.